app/models/hold.rb

```diff
@@ -19,22 +19,21 @@ class Hold < ActiveRecord::Base
   def handle_conflicting_rentals
     # if rental falls between data range of our hold
     # only reserved but not picked up
-    conflicting_rentals = Rental.reserved.where('start_time <= :hold_end_time AND end_time >= :hold_start_time AND item_id = :hold_item_id',
-                                                hold_start_time: start_time, hold_end_time: end_time, hold_item_id: item.id)
+    conflicting_rentals = Rental.reserved.joins('INNER JOIN rentals_items ON rentals_items.rental_id = rentals.id')
+                                .where('start_time <= :hold_end_time AND end_time >= :hold_start_time AND rentals_items.item_id = :hold_item_id',
+                                       hold_start_time: start_time, hold_end_time: end_time, hold_item_id: item.id)
     conflicting_rentals.each { |r| replace_rental(r) } unless conflicting_rentals.empty?
     start_hold
   end
 
   def replace_rental(curr_rental)
-    new_rental = Rental.new(item_type_id: curr_rental.item_type_id, creator_id: curr_rental.creator_id,
-                            renter_id: curr_rental.renter_id,
-                            start_time: curr_rental.start_time, end_time: curr_rental.end_time)
-    new_rental.create_reservation
+    new_rental = Rental.new(rentals_items_attributes: (curr_rental.item_types.map { |x| { item_type: x } }), creator_id: curr_rental.creator_id,
+                            renter_id: curr_rental.renter_id, start_time: curr_rental.start_time, end_time: curr_rental.end_time)
```

Comment:
If `creator` and `renter` are named associations, you should be able to reference them without the ID: `creator: curr_rental.creator, renter: curr_rental.renter`.